azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,954 @@
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+								},
+							},
+						},
+						"fixed_date": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MinItems:      1,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.recurrence"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"time_zone": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+									},
+									"start": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: rfc3339String(),
+									},
+									"end": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: rfc3339String(),
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+						"recurrence": {
+							Type:          schema.TypeSet,
+							Optional:      true,
+							MinItems:      1,
+							MaxItems:      1,
+							ConflictsWith: []string{"profile.fixed_date"},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"frequency": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(insights.Week),
+										}, false),
+									},
+									"schedule": {
+										Type:     schema.TypeSet,
+										Required: true,
+										MinItems: 1,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"time_zone": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(listTimeZoneNames(), false),
+												},
+												"days": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeString,
+													},
+												},
+												"hours": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeInt,
+													},
+												},
+												"minutes": {
+													Type:     schema.TypeList,
+													Required: true,
+													Elem: &schema.Schema{
+														Type: schema.TypeInt,
+													},
+												},
+											},
+										},
+										Set: resourceAzureRmAutoscaleDefaultHash,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"notification": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"operation": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Scale"}, false),
```

Comment:
(as above) can we make this case insensitive?

also: is there an Enum available for this?